Assets/Scripts/UI/HealthEffectController.cs

```diff
@@ -0,0 +1,134 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Rendering.Universal;
+
+public class HealthEffectController : MonoBehaviour
+{
+   
+   public Volume volume;
+
+   private Vignette vignette;
+   public bool vignetteToggle;
+
+   private FilmGrain filmgrain;
+   public bool filmgrainToggle;
+
+   private PaniniProjection paniniprojection;
+   public bool distanceToggle;
+   public bool cropToFitToggle;
+
+   private LensDistortion lensdistortion;
+   public bool lensDistortionToggle;
+
+   private Bloom bloom;
+   public bool bloomToggle;
+
+   public Killable killable;
+
+   void Start()
+   {
+
+    if(volume.profile.TryGet<Vignette>(out vignette))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<FilmGrain>(out filmgrain))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<PaniniProjection>(out paniniprojection))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<LensDistortion>(out lensdistortion))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+    if(volume.profile.TryGet<Bloom>(out bloom))
+    {
+        killable.OnHealthChange.AddListener(OnHealthChange);
+    }
+
+   }
+
+
+   void OnHealthChange()
+   {
+    UpdateVisualEffect(killable.currentHP);
+   }
+
+
+   void UpdateVisualEffect(float health)
+   {
+
+    float intensityController = Mathf.Clamp01(1.0f - (health / 100.0f));
```

Comment:
For each of these effects it should be a percentage of a value in between two values. For each of them they should have a start and max value, then calculate a value in between using the percentage of the health, like `currentHealth/maxHealth`. 

So the code would look like this:
`float newValue = Mathf.lerp(startValue, maxValue, currentHealth/maxHealth)`

Then you should apply the value using `Override()` since it allows you to go above the max 1 instensity, like `vignette.intensity.Override(newValue)`